lib/pos_blockchain.go

```diff
@@ -322,6 +339,21 @@ func (bc *Blockchain) validateAndIndexBlockPoS(block *MsgDeSoBlock) (*BlockNode,
 		// We return the block node as-is as a best effort thing.
 		return blockNode, nil
 	}
+	currentEpochEntry, err := utxoView.GetCurrentEpochEntry()
```

Comment:
Assuming I haven't missed anything, do we need to first verify here that the block belongs in the current epoch, before validating the partial signature? Even though `validateAndIndexBlockPoS` may currently only be called for blocks in the current epoch, it may be worth a check here again to bullet proof things.

Reply:
yes - if the current block is not in the current epoch after successfully constructing the view based on the parent block, then we'll never accept this block.